i2c/triac.py

```diff
@@ -0,0 +1,407 @@
+                                      smooth_change_in_progress=False,
+                                      thread=None, stop_thread=False)
+
+        self.log.info("Configued TriacDimmer %s: channel %d, mains frequency %dHz, PWM %s%%",
+                      self.name, self.channel, self.freq, self.state.initial)
+        self.log.debug("%s has following configured connections: \n%s",
+                       self.name, yaml.dump(self.comm))
+
+        # publish inital state back to remote connections
+        self.publish_actuator_state()
+
+        configure_device_channel(self.comm, is_output=False,
+                                 name="set duty cycle", datatype=ChanType.INTEGER,
+                                 unit="%", restrictions="0:100")
+        #the actuator gets registered twice, at core-actuator and here
+        # currently this is the only way to pass the device_channel_config to homie_conn
+        self._register(self.comm, None)
+
+    def on_message(self, msg):
+        """Called when the actuator receives a message.
+        Sets the triac according to the message value
+        """
+        self.log.debug("%s triac channel %d received command %s",
+                           self.name, self.channel, msg)
+
+        if msg.isdigit():
+            #msg contrains digits convert it from string to int
+            value = int(msg)
+        elif msg == "ON":
+            #handle openHab item sending ON
+            value = 100
+        elif msg == "OFF":
+            #handle openHab item sending OFF
+            value = 0
+        elif msg == "DIM":
+            #ignore the DIM command if smooth_change_in_progress
+            if not self.dimmer.smooth_change_in_progress:
+                if isinstance(self.dimmer.thread, Thread):
+                    #Info: waiting for the thread will rais toggle time for local connections,
+                    # so there might occure no short toggle event.
+                    # But there should be no running thread anyway
+                    self.dimmer.thread.join(timeout=None)
+                self.dimmer.stop_thread = False
+                #set target_value to 0 if current state > 0
+                if self.state.current:
+                    target_value = 0
+                else:
+                    target_value = 100
+                #start manual dimming with start_delay
+                self.dimmer.thread = Thread(target=self._smooth_dimmer,
+                                            args=(self.dimmer.dim_delay,
+                                                  self.dimmer.dim_interval, target_value))
+                #  _smooth_dimmer args= start_delay, interval_time, target_value
+                self.state.before_dimming = self.state.current
+                self.dimmer.thread.start()
+
+            return
+        elif msg == "STOP":
+            #make sure we don't interrupt the smooth change
+            if not self.dimmer.smooth_change_in_progress:
```

Comment:
`_smoot_dimmer()` is called in two cases:

1. The actuator receives ON / OFF or a nummer e.g. 55
2. The actuator receives DIM

In the first case, the actuator will smoothly change the PWM to 55%. To avoid interruption by DIM and STOP commands, `self.dimmer.smooth_change_in_progress` is checked beforehand. This allows the actuator to be connected to a RpiGPIOSensor. `self.dimmer.smooth_change_in_progress` is initialized to False in `__init__()`.

If `self.dimmer.stop_thread` is never set to True, `_smoot_dimmer()` will run until the target_value is reached e. g. 100 and exit. `self.dimmer.stop_thread` is only needed to exit early.

>All of these flags setting are spread throughout the code which makes me nervous. Is there a way to create transactions somehow or at least move the setting to true and false of theses flags closer together? 

I'm not sure how to archive this. Should I refractor the if statement below to make the structure more clearer?
```python
if msg.isdigit():
   #code
elif msg == "ON":
   #code
elif msg == "OFF":
   #code
elif msg == "DIM":
   #code
elif msg == "STOP":
   #code
elif is_toggle_cmd(msg):
   #code
else:
   #code
```

Thank you for reviewing the code! =)